#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "CASTER"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.22305897466414973 -0.8813237723991194 -0.41654903916101327 2.2975837681646563
  position -0.7946108292522793 2.1423268310243597 -1.2948571548442578
}
TexturedBackground {
}
TexturedBackgroundLight {
  luminosity 0.3
}
Floor {
  translation 1 0 -2.5
  size 4 6
  appearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
}
DEF Light SpotLight {
  attenuation 0 0 1
  beamWidth 1.5
  cutOffAngle 1.57
  direction 0 -1 0
  intensity 10
  location 1 3.8 -2.5
  radius 10
  castShadows TRUE
}
DEF Arena_walls Solid {
  rotation 1 0 0 -1.5707953071795862
  scale 0.01 0.01 0.01
  children [
    DEF WALLS Shape {
      appearance PBRAppearance {
        baseColor 0.666667 0.333333 0
        roughness 0.9
        metalness 0
      }
      geometry Mesh {
        url [
          "wall1.stl"
        ]
      }
    }
  ]
  name "WALLS"
  boundingObject USE WALLS
}
DEF Arena_track Solid {
  rotation 1 0 0 -1.5707953071795862
  scale 0.01 0.01 0.01
  children [
    DEF TRACK Shape {
      appearance PBRAppearance {
        roughness 0.9
        metalness 0
      }
      geometry Mesh {
        url [
          "samplearena3.stl"
        ]
      }
    }
  ]
  name "TRACK"
}
Robot {
  translation 0.19442 0.0548906 -0.13645
  rotation 0.00023065097392841868 0.9999998869652361 0.00041577595300285806 1.5708
  children [
    DEF color_detection Transform {
      children [
        Solid {
          translation 0 0.0725882 -0.009
          rotation 1 0 0 0.295
          children [
            Camera {
              rotation 1 0 0 1.57
              name "camera1"
              fieldOfView 0.2
              width 8
              height 8
            }
            DEF colorSensor Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 0
                roughness 0.5
                metalness 0
              }
              geometry Box {
                size 0.02 0.005 0.02
              }
            }
          ]
          name "colorSensor1"
          boundingObject USE colorSensor
          physics Physics {
            mass 0.01
          }
        }
        Solid {
          translation 0 0.116 -0.015
          rotation 1 0 0 2.29
          children [
            Camera {
              rotation 1 0 0 1.57
              name "camera2"
              fieldOfView 0.2
              width 8
              height 8
            }
            DEF colorSensor Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 0
                roughness 0.5
                metalness 0
              }
              geometry Box {
                size 0.02 0.005 0.02
              }
            }
          ]
          name "colorSensor1(1)"
          boundingObject USE colorSensor
          physics Physics {
            mass 0.01
          }
        }
        Solid {
          translation 0 0.08 -0.03
          rotation 1 0 0 -1.5707996938995747
          children [
            DEF colorSensors Shape {
              appearance Appearance {
                material Material {
                  ambientIntensity 1
                  diffuseColor 0.792157 0.819608 0.933333
                  shininess 0.4
                  specularColor 0.396078 0.409804 0.466667
                }
              }
              geometry IndexedFaceSet {
                coord Coordinate {
                  point [
                    -0.015 -0.031 -0.024544
                    -0.015 -0.031 -0.011193
                    -0.015 -0.02273 0.044549
                    -0.015 -0.019627 0.047233
                    -0.015 -0.003103 -0.002869
                    -0.015 -0.003103 0.02186
                    -0.015 0 -0.024544
                    -0.015 0 0.024544
                    0.015 -0.031 -0.024544
                    0.015 -0.031 -0.011193
                    0.015 -0.02273 0.044549
                    0.015 -0.019627 0.047233
                    0.015 -0.003103 -0.002869
                    0.015 -0.003103 0.02186
                    0.015 0 -0.024544
                    0.015 0 0.024544
                  ]
                }
                normal Normal {
                  vector [
                    -1 0 0
                    0 -1 0
                    0 -0.756296 -0.654229
                    0 -0.654229 0.756296
                    0 -0.28594 0.958248
                    0 0 -1
                    0 0.756296 0.654229
                    0 1 0
                    1 0 0
                  ]
                }
                coordIndex [
                  1, 0, 9, -1, 9, 0, 8, -1, 12, 9, 8, -1, 15, 11, 10, -1, 10, 13, 15, -1, 15, 13, 12, -1, 15, 12, 14, -1, 14, 12, 8, -1, 15, 14, 7, -1, 7, 14, 6, -1, 2, 3, 5, -1, 5, 3, 7, -1, 5, 7, 4, -1, 4, 7, 6, -1, 4, 6, 1, -1, 1, 6, 0, -1, 6, 14, 0, -1, 0, 14, 8, -1, 1, 9, 4, -1, 4, 9, 12, -1, 5, 4, 13, -1, 13, 4, 12, -1, 3, 11, 7, -1, 7, 11, 15, -1, 10, 2, 13, -1, 13, 2, 5, -1, 2, 10, 3, -1, 3, 10, 11, -1
                ]
                normalIndex [
                  1, 1, 1, -1, 1, 1, 1, -1, 8, 8, 8, -1, 8, 8, 8, -1, 8, 8, 8, -1, 8, 8, 8, -1, 8, 8, 8, -1, 8, 8, 8, -1, 7, 7, 7, -1, 7, 7, 7, -1, 0, 0, 0, -1, 0, 0, 0, -1, 0, 0, 0, -1, 0, 0, 0, -1, 0, 0, 0, -1, 0, 0, 0, -1, 5, 5, 5, -1, 5, 5, 5, -1, 4, 4, 4, -1, 4, 4, 4, -1, 1, 1, 1, -1, 1, 1, 1, -1, 6, 6, 6, -1, 6, 6, 6, -1, 2, 2, 2, -1, 2, 2, 2, -1, 3, 3, 3, -1, 3, 3, 3, -1
                ]
              }
            }
          ]
          name "color_sensors"
          boundingObject USE colorSensors
          physics Physics {
            mass 0.05
            centerOfMass [
              0 0 0
            ]
          }
        }
      ]
    }
    InertialUnit {
      translation 0.05 0 0
      rotation 0 1 0 -1.57
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.001 0.01
              }
            }
          ]
        }
      ]
      name "inertial_unit"
      resolution 0.001
    }
    DEF IR_PANEL Transform {
      translation 0.0045 -0.045 0.03
      children [
        Solid {
          translation 0.03 0.0225 0.004
          children [
            Shape {
              geometry Cylinder {
                height 0.045
                radius 0.001
              }
            }
          ]
          name "spacer"
        }
        Solid {
          translation 0.03 0.0225 -0.004
          children [
            Shape {
              geometry Cylinder {
                height 0.045
                radius 0.001
              }
            }
          ]
          name "spacer(1)"
        }
        Solid {
          translation -0.03 0.0225 0.004
          children [
            Shape {
              geometry Cylinder {
                height 0.045
                radius 0.001
              }
            }
          ]
          name "spacer(2)"
        }
        Solid {
          translation -0.03 0.0225 -0.004
          children [
            Shape {
              geometry Cylinder {
                height 0.045
                radius 0.001
              }
            }
          ]
          name "spacer(3)"
        }
        DistanceSensor {
          translation 0.0334 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr0"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
          aperture 1
        }
        DistanceSensor {
          translation 0.0239 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr1"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        DistanceSensor {
          translation 0.0144 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr2"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        DistanceSensor {
          translation 0.0049 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr3"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        DistanceSensor {
          translation -0.0056 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr4"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        DistanceSensor {
          translation -0.0151 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr5"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        DistanceSensor {
          translation -0.0246 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr6"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        DistanceSensor {
          translation -0.0341 -0.001 0
          rotation 0 0 1 -1.57
          children [
            Solid {
              translation 0 -0.001 0
              rotation 0 0 1 -1.57
              children [
                Shape {
                  geometry Box {
                    size 0.003 0.001 0.003
                  }
                }
              ]
              name "qtr_body"
            }
          ]
          name "qtr7"
          lookupTable [
            0 0 0.01
            0.04 1000 0.01
          ]
          type "infra-red"
        }
        Solid {
          children [
            DEF ip_PCB Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.075 0.00157 0.0127
              }
            }
          ]
          name "sensor_panel_PCB"
          boundingObject USE ip_PCB
          physics Physics {
            density -1
            mass 0.005
          }
        }
      ]
    }
    DEF MOTOR_DRIVER Solid {
      translation 0 0.005 0
      children [
        DEF driver Shape {
          appearance Appearance {
            material Material {
            }
          }
          geometry Box {
            size 0.065 0.01 0.065
          }
        }
      ]
      name "solid(4)"
      boundingObject USE driver
      physics Physics {
        density -1
        mass 0.05
      }
    }
    DEF BATTERY Solid {
      translation -1.10196e-07 0.01 -0.06
      rotation 0 1 0 1.5708
      children [
        DEF BATTERY Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
          }
          geometry Box {
            size 0.034 0.025 0.106
          }
        }
      ]
      name "solid(4)"
      boundingObject USE BATTERY
      physics Physics {
        density -1
        mass 0.249
      }
    }
    DEF motor_left Solid {
      translation 0.033 -0.0225 -0.03654
      rotation 0 0 1 -1.57
      children [
        DEF motor Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.064
            radius 0.0125
            subdivision 18
          }
          castShadows FALSE
        }
      ]
      boundingObject USE motor
      physics DEF motor_physics Physics {
        density -1
        mass 0.101
        centerOfMass [
          0 0 0
        ]
      }
    }
    DEF motor_right Solid {
      translation -0.033 -0.0225 -0.0365
      rotation 0 0 1 1.57
      children [
        USE motor
      ]
      name "solid(4)"
      boundingObject USE motor
      physics USE motor_physics
    }
    DEF Upper_Platform Solid {
      translation 0 0.052 -0.036
      children [
        DEF Power_Supply Solid {
          translation -0.05301 0.015 0
          children [
            DEF power_supply Shape {
              geometry Box {
                size 0.048 0.025 0.065
              }
            }
          ]
          name "solid(1)"
          boundingObject USE power_supply
          physics Physics {
            density -1
            mass 0.173
          }
        }
        DEF Arduino Solid {
          translation 0.04847 0.007605 0
          children [
            DEF arduino Shape {
              geometry Box {
                size 0.05342 0.01021 0.1016
              }
            }
          ]
          boundingObject USE arduino
          physics Physics {
            density -1
            mass 0.037
          }
        }
        DEF upper_Layer Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.17 0.005 0.115
          }
        }
      ]
      name "solid(7)"
      boundingObject USE upper_Layer
      physics Physics {
        mass 0.05
        centerOfMass [
          0 0 0
        ]
      }
      recognitionColors [
        1 0 0
      ]
    }
    DEF Spacer_4 Solid {
      translation 0.08 0.025 0.009
      children [
        DEF spacer Shape {
          geometry Cylinder {
            height 0.05
            radius 0.004
          }
        }
      ]
      name "solid(6)"
      boundingObject USE spacer
      physics DEF spacer Physics {
        density -1
        mass 0.005
      }
    }
    DEF Spacer_3 Solid {
      translation -0.08 0.025 0.009
      children [
        DEF spacer Shape {
          geometry Cylinder {
            height 0.05
            radius 0.004
          }
        }
      ]
      name "solid(5)"
      boundingObject USE spacer
      physics Physics {
        density -1
        mass 0.005
      }
    }
    DEF Spacer_2 Solid {
      translation -0.08 0.025 -0.08
      children [
        DEF spacer Shape {
          geometry Cylinder {
            height 0.05
            radius 0.004
          }
        }
      ]
      name "solid(4)"
      boundingObject USE spacer
      physics Physics {
        density -1
        mass 0.005
      }
    }
    DEF Spacer_1 Solid {
      translation 0.08 0.025 -0.08
      children [
        DEF spacer Shape {
          geometry Cylinder {
            height 0.05
            radius 0.004
          }
        }
      ]
      name "solid(5)"
      boundingObject USE spacer
      physics Physics {
        density -1
        mass 0.005
      }
    }
    DistanceSensor {
      translation 0 -0.015 0.0815
      rotation 0 1 0 -1.57
      children [
        DEF IR Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.01 0.01 0.025
          }
        }
      ]
      name "front_ir"
      boundingObject DEF IR Shape {
        geometry Box {
          size 0.01 0.01 0.025
        }
      }
      physics DEF SHARP_IR Physics {
        density -1
        mass 0.01
      }
    }
    DEF CASTER Solid {
      translation 0 -0.025 0.06
      children [
        Solid {
          translation 0 -0.018 0
          children [
            DEF caster_ball Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.01375
                subdivision 2
              }
            }
          ]
          name "solid(1)"
          contactMaterial "CASTER"
          boundingObject USE caster_ball
          physics Physics {
            density -1
            mass 0.005
          }
        }
        Solid {
          children [
            DEF caster_cylinder Shape {
              appearance PBRAppearance {
                baseColor 0 0 0
              }
              geometry Cylinder {
                height 0.035
                radius 0.015
              }
            }
          ]
          boundingObject USE caster_cylinder
          physics Physics {
            density -1
            mass 0.02
          }
        }
      ]
      name "solid(2)"
      contactMaterial "CASTER"
      physics Physics {
      }
    }
    DEF Servo_block1 Solid {
      translation 0.03795 0.01 0.05036
      children [
        DEF Servo1_box Shape {
          geometry Box {
            size 0.03721 0.01816 0.031
          }
        }
      ]
      name "solid(3)"
      boundingObject USE Servo1_box
      physics Physics {
        density -1
        mass 0.05
      }
    }
    DEF BOT_LAYER Solid {
      children [
        DEF BOT_LAYER Shape {
          appearance Appearance {
            material Material {
              ambientIntensity 1
              diffuseColor 0 0.819608 0
              shininess 0.4
              specularColor 0.396078 0.409804 0.466667
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.085 -0.01 -0.080353, -0.085 -0.01 -0.075353, -0.085 -0.01 -0.005393, -0.085 -0.01 0.072647, -0.085 0 -0.080353, -0.085 0 -0.075353, -0.085 0 -0.005393, -0.085 0 0.072647, -0.084659 -0.01 -0.082941, -0.084659 -0.01 0.075235, -0.084659 0 -0.082941, -0.084659 0 0.075235, -0.08366 -0.01 -0.085353, -0.08366 -0.01 0.077647, -0.08366 0 -0.085353, -0.08366 0 0.077647, -0.082071 -0.01 -0.087424, -0.082071 -0.01 0.079718, -0.082071 0 -0.087424, -0.082071 0 0.079718, -0.08 -0.01 -0.089013, -0.08 -0.01 0.081307, -0.08 0 -0.089013, -0.08 0 0.081307, -0.077588 -0.01 -0.090012, -0.077588 -0.01 0.082306, -0.077588 0 -0.090012, -0.077588 0 0.082306, -0.075 -0.01 -0.090353, -0.075 -0.01 0.082647, -0.075 0 -0.090353, -0.075 0 0.082647, -0.065 -0.01 -0.075353, -0.065 -0.01 -0.005393, -0.065 0 -0.075353, -0.065 0 -0.005393, 0.065 -0.01 -0.075353, 0.065 -0.01 -0.005393, 0.065 0 -0.075353, 0.065 0 -0.005393, 0.075 -0.01 -0.090353, 0.075 -0.01 0.082647, 0.075 0 -0.090353, 0.075 0 0.082647, 0.077588 -0.01 -0.090012, 0.077588 -0.01 0.082306, 0.077588 0 -0.090012, 0.077588 0 0.082306, 0.08 -0.01 -0.089013, 0.08 -0.01 0.081307, 0.08 0 -0.089013, 0.08 0 0.081307, 0.082071 -0.01 -0.087424, 0.082071 -0.01 0.079718, 0.082071 0 -0.087424, 0.082071 0 0.079718, 0.08366 -0.01 -0.085353, 0.08366 -0.01 0.077647, 0.08366 0 -0.085353, 0.08366 0 0.077647, 0.084659 -0.01 -0.082941, 0.084659 -0.01 0.075235, 0.084659 0 -0.082941, 0.084659 0 0.075235, 0.085 -0.01 -0.080353, 0.085 -0.01 -0.075353, 0.085 -0.01 -0.005393, 0.085 -0.01 0.072647, 0.085 0 -0.080353, 0.085 0 -0.075353, 0.085 0 -0.005393, 0.085 0 0.072647
              ]
            }
            normal Normal {
              vector [
                -1 0 0, -0.965926 0 -0.258819, -0.965926 0 0.258819, -0.866025 0 -0.5, -0.866025 0 0.5, -0.707107 0 -0.707107, -0.707107 0 0.707107, -0.5 0 -0.866025, -0.5 0 0.866025, -0.258819 0 -0.965926, -0.258819 0 0.965926, 0 -1 0, 0 0 -1, 0 0 1, 0 1 0, 0.258819 0 -0.965926, 0.258819 0 0.965926, 0.5 0 -0.866025, 0.5 0 0.866025, 0.707107 0 -0.707107, 0.707107 0 0.707107, 0.866025 0 -0.5, 0.866025 0 0.5, 0.965926 0 -0.258819, 0.965926 0 0.258819, 1 0 0
              ]
            }
            coordIndex [
              71, 70, 39, -1, 47, 51, 71, -1, 38, 34, 39, -1, 39, 34, 35, -1, 51, 55, 71, -1, 71, 55, 59, -1, 71, 59, 63, -1, 47, 71, 43, -1, 43, 71, 39, -1, 43, 39, 31, -1, 31, 39, 35, -1, 31, 35, 6, -1, 46, 42, 38, -1, 38, 42, 30, -1, 38, 30, 34, -1, 34, 30, 26, -1, 69, 68, 38, -1, 38, 68, 62, -1, 38, 62, 58, -1, 58, 54, 38, -1, 38, 54, 50, -1, 38, 50, 46, -1, 26, 22, 34, -1, 34, 22, 18, -1, 34, 18, 14, -1, 14, 10, 34, -1, 34, 10, 4, -1, 34, 4, 5, -1, 6, 7, 31, -1, 31, 7, 11, -1, 31, 11, 15, -1, 15, 19, 31, -1, 31, 19, 23, -1, 31, 23, 27, -1, 2, 33, 3, -1, 3, 33, 29, -1, 32, 36, 33, -1, 33, 36, 37, -1, 33, 37, 29, -1, 37, 66, 29, -1, 29, 66, 67, -1, 29, 67, 41, -1, 41, 67, 61, -1, 41, 61, 57, -1, 21, 17, 13, -1, 57, 53, 41, -1, 41, 53, 49, -1, 41, 49, 45, -1, 29, 25, 3, -1, 3, 25, 21, -1, 3, 21, 9, -1, 9, 21, 13, -1, 24, 28, 32, -1, 32, 28, 40, -1, 32, 40, 36, -1, 36, 40, 44, -1, 1, 0, 32, -1, 32, 0, 8, -1, 32, 8, 12, -1, 12, 16, 32, -1, 32, 16, 20, -1, 32, 20, 24, -1, 44, 48, 36, -1, 36, 48, 52, -1, 36, 52, 56, -1, 56, 60, 36, -1, 36, 60, 64, -1, 36, 64, 65, -1, 29, 41, 31, -1, 31, 41, 43, -1, 37, 36, 39, -1, 39, 36, 38, -1, 69, 38, 65, -1, 65, 38, 36, -1, 68, 69, 64, -1, 64, 69, 65, -1, 40, 28, 42, -1, 42, 28, 30, -1, 0, 1, 4, -1, 4, 1, 5, -1, 34, 5, 32, -1, 32, 5, 1, -1, 35, 34, 33, -1, 33, 34, 32, -1, 7, 3, 9, -1, 7, 9, 11, -1, 11, 9, 13, -1, 11, 13, 15, -1, 15, 13, 17, -1, 15, 17, 19, -1, 19, 17, 21, -1, 19, 21, 23, -1, 23, 21, 25, -1, 23, 25, 27, -1, 27, 25, 29, -1, 27, 29, 31, -1, 43, 41, 45, -1, 43, 45, 47, -1, 47, 45, 49, -1, 47, 49, 51, -1, 51, 49, 53, -1, 51, 53, 55, -1, 55, 53, 57, -1, 55, 57, 59, -1, 59, 57, 61, -1, 59, 61, 63, -1, 63, 61, 67, -1, 63, 67, 71, -1, 68, 64, 60, -1, 68, 60, 62, -1, 62, 60, 56, -1, 62, 56, 58, -1, 58, 56, 52, -1, 58, 52, 54, -1, 54, 52, 48, -1, 54, 48, 50, -1, 50, 48, 44, -1, 50, 44, 46, -1, 46, 44, 40, -1, 46, 40, 42, -1, 30, 28, 24, -1, 30, 24, 26, -1, 26, 24, 20, -1, 26, 20, 22, -1, 22, 20, 16, -1, 22, 16, 18, -1, 18, 16, 12, -1, 18, 12, 14, -1, 14, 12, 8, -1, 14, 8, 10, -1, 10, 8, 0, -1, 10, 0, 4, -1, 2, 3, 6, -1, 6, 3, 7, -1, 33, 2, 35, -1, 35, 2, 6, -1, 66, 37, 70, -1, 70, 37, 39, -1, 67, 66, 71, -1, 71, 66, 70, -1
            ]
            normalIndex [
              14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 14, 14, 14, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 11, 11, 11, -1, 13, 13, 13, -1, 13, 13, 13, -1, 25, 25, 25, -1, 25, 25, 25, -1, 13, 13, 13, -1, 13, 13, 13, -1, 25, 25, 25, -1, 25, 25, 25, -1, 12, 12, 12, -1, 12, 12, 12, -1, 0, 0, 0, -1, 0, 0, 0, -1, 13, 13, 13, -1, 13, 13, 13, -1, 0, 0, 0, -1, 0, 0, 0, -1, 0, 0, 2, -1, 0, 2, 2, -1, 2, 2, 4, -1, 2, 4, 4, -1, 4, 4, 6, -1, 4, 6, 6, -1, 6, 6, 8, -1, 6, 8, 8, -1, 8, 8, 10, -1, 8, 10, 10, -1, 10, 10, 13, -1, 10, 13, 13, -1, 13, 13, 16, -1, 13, 16, 16, -1, 16, 16, 18, -1, 16, 18, 18, -1, 18, 18, 20, -1, 18, 20, 20, -1, 20, 20, 22, -1, 20, 22, 22, -1, 22, 22, 24, -1, 22, 24, 24, -1, 24, 24, 25, -1, 24, 25, 25, -1, 25, 25, 23, -1, 25, 23, 23, -1, 23, 23, 21, -1, 23, 21, 21, -1, 21, 21, 19, -1, 21, 19, 19, -1, 19, 19, 17, -1, 19, 17, 17, -1, 17, 17, 15, -1, 17, 15, 15, -1, 15, 15, 12, -1, 15, 12, 12, -1, 12, 12, 9, -1, 12, 9, 9, -1, 9, 9, 7, -1, 9, 7, 7, -1, 7, 7, 5, -1, 7, 5, 5, -1, 5, 5, 3, -1, 5, 3, 3, -1, 3, 3, 1, -1, 3, 1, 1, -1, 1, 1, 0, -1, 1, 0, 0, -1, 0, 0, 0, -1, 0, 0, 0, -1, 12, 12, 12, -1, 12, 12, 12, -1, 12, 12, 12, -1, 12, 12, 12, -1, 25, 25, 25, -1, 25, 25, 25, -1
            ]
          }
        }
      ]
      name "solid(1)"
      boundingObject USE BOT_LAYER
      physics Physics {
        density -1
        mass 0.25
      }
      recognitionColors [
        0 0 0
      ]
    }
    DEF LEFT HingeJoint {
      jointParameters HingeJointParameters {
        position 42.11196320223189
        anchor 0.0775 -0.0225 -0.0403
      }
      device [
        RotationalMotor {
          name "left_motor"
        }
      ]
      endPoint DEF WHEEL_LEFT Solid {
        translation 0.0775 -0.022547773694616722 -0.040285246064419106
        rotation 0.627227377454589 -0.6267280982557457 -0.4623826422219748 4.008406486289448
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.025
              radius 0.0325
            }
          }
        ]
        name "left_wheel"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
          density -1
          mass 0.1
        }
      }
    }
    DEF RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        position 16.512021051021286
        anchor -0.0775 -0.0225 -0.0403
      }
      device [
        RotationalMotor {
          name "right_motor"
        }
      ]
      endPoint Solid {
        translation -0.0775 -0.02249999997380849 -0.04030000002523184
        rotation 0.6774556601219112 -0.6769163987130592 -0.2878159441763579 3.7024993626131493
        children [
          USE WHEEL
        ]
        name "right_wheel"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
  ]
  boundingObject USE BOT_LAYER
  physics Physics {
  }
  controller "fF"
}
